Source/Python/WarpXWrappers.cpp

```diff
@@ -494,6 +495,72 @@ extern "C"
         mypc.defineAllParticleTiles();
     }
 
+    int warpx_getParticleBoundaryBufferSize(const char* species_name, int boundary)
+    {
+        const std::string name(species_name);
+        auto& particle_buffers = WarpX::GetInstance().GetParticleBoundaryBuffer();
+        return particle_buffers.getNumParticlesInContainer(species_name, boundary);
+    }
+
+    int** warpx_getParticleBoundaryBufferScrapedSteps(const char* species_name, int boundary, int lev,
+                     int* num_tiles, int** particles_per_tile)
+    {
+        const std::string name(species_name);
+        auto& particle_buffers = WarpX::GetInstance().GetParticleBoundaryBuffer();
+        auto& particle_buffer = particle_buffers.getParticleBuffer(species_name, boundary);
+
+        const int comp = particle_buffer.NumIntComps() - 1;
```

Comment:
Assuming this is always the last integer component seems pretty brittle.

Would it make more sense to make this something like `warpx_getParticleBoundaryBufferInt` with a flexible component name? It would mirror the functionality of real components that way...

Reply:
Unfortunately since the particle buffer is a `amrex::AmrParticleContainer` and not a `warpx::ParticleContainer` it doesn't support named PIDs. However, this specific integer component is added to the particle buffer after all the other PIDs have been copied over (from whatever `warpx::ParticleContainer` the buffer gets its particles) so this 'last integer component' approach can't be broken by the user at runtime. That being said I agree this should be made more durable. The best approach is probably to save the index of this component for each particle buffer in an array or vector when it is created. I'll make a new card for that on Trello and we can make that change in an upstream PR.

Reply:
Alternatively, you can bring up the issue in Andrew's PR and maybe he'll just fix it before that PR is merged.

Reply:
There is actually an extended problem here where any runtime attribute added to a species' particle container after the buffer for that particle (for a given boundary) has been created won't be included in the buffer. So far I don't think there is a specific case where this will be a problem for us since we generally add all the extra PID's before the first step of the simulation.